backend/benefit/applications/api/v1/application_views.py

```diff
@@ -394,11 +394,25 @@ class Meta:
             "recovery_amount",
         ]
 
+
+class ApplicantApplicationAlterationViewSet(BaseApplicationAlterationViewSet):
+    permission_classes = [BFIsApplicant, TermsOfServiceAccepted]
+
+    APPLICANT_UNEDITABLE_FIELDS = [
+        "state",
+        "recovery_start_date",
+        "recovery_end_date",
+        "recovery_amount",
+        "recovery_justification",
+        "is_recoverable",
+        "handled_at",
+        "handled_by",
+    ]
+
     def _prune_fields(self, request):
-        if not request.user.is_handler():
-            for field in self.APPLICANT_UNEDITABLE_FIELDS:
-                if field in request.data.keys():
-                    request.data.pop(field)
+        for field in self.APPLICANT_UNEDITABLE_FIELDS:
```

Comment:
I haven't seen this pattern before. Isn't there `read_only_fields` for this?

https://www.django-rest-framework.org/api-guide/serializers/#specifying-read-only-fields

Reply:
That could possibly suffice now that the handler check is no longer a part of the function, yeah. I'll check whether that's the case.